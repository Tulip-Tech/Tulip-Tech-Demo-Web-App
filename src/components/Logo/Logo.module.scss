@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

//
// Logo
// --------------------------------

.logo {
  width: auto;
  max-height: variables.$header-height - 10px;
  vertical-align: middle;
  cursor: pointer;
}

//
// mediaQueries
// --------------------------------

@include responsive.mobile-and-tablet() {
  .logo {
    max-height: variables.$header-height-mobile - 20px;
  }
}
