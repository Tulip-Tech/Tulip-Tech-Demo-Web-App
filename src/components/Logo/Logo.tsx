import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import styles from './Logo.module.scss';

type Props = {
  src: string;
  onLoad: () => void;
};

type ImgRef = {
  height?: number;
  width?: number;
};

const Logo: React.FC<Props> = ({ src, onLoad }: Props) => {
  const [imgDimensions, updateImgDimensions] = useState<ImgRef>({ height: undefined, width: undefined });

  const onLoadHandler = (event: React.SyntheticEvent<HTMLImageElement, Event>) => {
    const { height, width } = event.currentTarget;
    updateImgDimensions({ height, width });
    onLoad();
  };

  return (
    <Link to="/">
      <img className={styles.logo} alt="logo" src={src} height={16} width={imgDimensions.width} onLoad={onLoadHandler} onError={onLoad} />
    </Link>
  );
};

export default Logo;
