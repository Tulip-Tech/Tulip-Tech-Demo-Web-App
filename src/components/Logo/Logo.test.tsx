import React from 'react';
import { fireEvent } from '@testing-library/react';

import Logo from './Logo';

import { renderWithRouter } from '#test/testUtils';

describe('<Logo/>', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderWithRouter(<Logo src="123" onLoad={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });

  it('calls the onLoad function after the image has been loaded', () => {
    const onLoad = vi.fn();
    const { getByAltText } = renderWithRouter(<Logo src="testimage" onLoad={onLoad} />);

    fireEvent.load(getByAltText('logo'));

    expect(onLoad).toBeCalledTimes(1);
  });

  it('calls the onLoad function when the image fails to load', () => {
    const onLoad = vi.fn();
    const { getByAltText } = renderWithRouter(<Logo src="testimage" onLoad={onLoad} />);

    fireEvent.error(getByAltText('logo'));

    expect(onLoad).toBeCalledTimes(1);
  });
});
