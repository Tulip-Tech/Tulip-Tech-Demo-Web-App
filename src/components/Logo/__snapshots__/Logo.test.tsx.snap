// Vitest Snapshot v1

exports[`<Logo/> > renders and matches snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="logo"
      class="_logo_f7d212"
      src="123"
    />
  </a>
</div>
`;
